internal/test/framework/framework.go

```diff
@@ -175,3 +177,14 @@ func (f *TestFramework) TearDown() {
 		}
 	}
 }
+
+// WriteToArtifactDir will write contents to $ARTIFACT_DIR/vmName/filename
+func (f *TestFramework) WriteToArtifactDir(contents []byte, vmName, filename string) error {
```

Comment:
Given this is a generic function, `vmName` should be `subDirName`. Thinking more about this I suggest allowing the caller to provide an empty `subDirName` giving us the option to not use a sub-dir if needed. #futureproof